libcaf_net/caf/net/http/client.test.cpp

```diff
@@ -0,0 +1,539 @@
+      client->add_header_field("Accept-Encoding", "chunked");
+      client->end_header();
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+  GIVEN("a multi-line HTTP request with a payload") {
+    auto expected = "POST /foo/bar/index.html HTTP/1.1\r\n"
+                    "Content-Length: 13\r\n"
+                    "Content-Type: plain/text\r\n"
+                    "\r\n"
+                    "Hello, world!"sv;
+    WHEN("the client sends the message") {
+      client->begin_header(net::http::method::post, "/foo/bar/index.html"sv);
+      client->add_header_field("Content-Length", "13");
+      client->add_header_field("Content-Type", "plain/text");
+      client->end_header();
+      client->send_payload(as_bytes(make_span("Hello, world!"sv)));
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+  GIVEN("a multi-line HTTP request with a chunked payload") {
+    auto expected = "POST /foo/bar/index.html HTTP/1.1\r\n"
+                    "Content-Type: plain/text\r\n"
+                    "Transfer-Encoding: chunked\r\n"
+                    "\r\n"
+                    "D\r\n"
+                    "Hello, world!\r\n"
+                    "11\r\n"
+                    "Developer Network\r\n"
+                    "0\r\n"
+                    "\r\n"sv;
+    WHEN("the client sends the message") {
+      client->begin_header(net::http::method::post, "/foo/bar/index.html"sv);
+      client->add_header_field("Content-Type", "plain/text");
+      client->add_header_field("Transfer-Encoding", "chunked");
+      client->end_header();
+      client->send_chunk(as_bytes(make_span("Hello, world!"sv)));
+      client->send_chunk(as_bytes(make_span("Developer Network"sv)));
+      client->send_end_of_chunks();
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+}
+
+OUTLINE("Sending all available HTTP methods") {
+  auto app_ptr = app_t::make();
+  auto client_ptr = net::http::client::make(std::move(app_ptr));
+  auto client = client_ptr.get();
+  auto transport = mock_stream_transport::make(std::move(client_ptr));
+  require_eq(transport->start(nullptr), error{});
+  GIVEN("a http request with <method> method") {
+    auto method_name = block_parameters<std::string>();
+    auto expected = detail::format("{} /foo/bar HTTP/1.1\r\n\r\n", method_name);
+    WHEN("the client sends the <enum value> message") {
+      auto method = static_cast<net::http::method>(block_parameters<uint8_t>());
+      client->begin_header(method, "/foo/bar"sv);
+      client->end_header();
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+  EXAMPLES = R"(
+    |  method  | enum value |
+    | GET      |     0      |
```

Comment:
`method` has `to_string` and `from_string`, so we can actually use proper names here:

```suggestion
    | GET      |     get      |
```

and then just `block_parameters<net::http::method>()` to get rid of the static casts.

The enum values are the C++ identifiers from the source file: https://github.com/actor-framework/actor-framework/blob/master/libcaf_net/caf/net/http/method.hpp.

Reply:
I tested the exact same setup before and got conversion errors when using `get` and `method::get`. After looking at the output from `to_string`, I got it working by using the full enum name, including the namespace. GET would be `caf::net::http::method::get`. Might be worth adding to the docs? Also might be worth looking into, an user would probably expect at least `method::get` to work.

Reply:
Yeah, I've realized this as well. See https://github.com/actor-framework/actor-framework/pull/1621. We can pull that in first and then rebase this branch.

Reply:
Sounds good. Reviewed it.